addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/HistoryProvider.java

```diff
@@ -61,21 +91,78 @@ public HttpMessage getHttpMessage(int historyId)
         return null;
     }
 
+    /**
+     * The query is ordered DESCending so the List and subsequent processing should be newest
+     * message first.
+     */
+    List<Integer> getMessageIds(int first, int last, String value) {
+        if (!server) {
+            if (Model.getSingleton().getDb().getDatabaseServer()
+                    instanceof ParosDatabaseServer pdbs) {
+                pds = pdbs;
+                server = true;
+            } else {
+                if (pds == null && !warnedNonParosDb) {
+                    LOGGER.warn("Unsupported Database Server.");
+                    warnedNonParosDb = true;
+                }
+            }
+        }
+        if (pds == null) {
+            return null;
+        }
+
+        Connection conn = null;
+        try {
+            conn = pds.getSingletonConnection();
+        } catch (SQLException | NullPointerException e) {
```

Comment:
The NPE will no longer happen, I'd inline the `pds.getSingletonConnection()` in the try-with-resource statement.